File: vite-vue-ts-frontend/src/modules/notes/types/dto.ts
import type { UserBaseResponse } from "../../users/types/dto";

export type AddRequest = {
  body: string;
};

export type NoteResponse = {
  id: string;
  user: UserBaseResponse;
  createdAt: number;
  updatedAt: number | null;
  body: string;
};

export type SearchResponse = {
  notes: NoteResponse[];
};

File: golang-backend/internal/handlers/notehandler/dto.go
package notehandler

import (
	"github.com/aportela/doneo/internal/handlers/userhandler"
)

type addRequest struct {
	Body string `json:"body"`
}

type NoteResponse struct {
	ID        string                       `json:"id"`
	User      userhandler.UserBaseResponse `json:"user"`
	CreatedAt int64                        `json:"createdAt"`
	UpdatedAt *int64                       `json:"updatedAt"`
	Body      string                       `json:"body"`
}

type searchResponse struct {
	ProjectPermissions []NoteResponse `json:"notes"`
}
